e2e/e2e_test.go

```diff
@@ -106,9 +112,84 @@ var _ = Describe("Whereabouts functionality", func() {
 
 			It("allocates a single pod within the correct IP range", func() {
 				By("checking pod IP is within whereabouts IPAM range")
-				secondaryIfaceIP, err := retrievers.SecondaryIfaceIPValue(pod)
+				secondaryIfaceIPs, err := retrievers.SecondaryIfaceIPValue(pod)
 				Expect(err).NotTo(HaveOccurred())
-				Expect(inRange(ipv4TestRange, secondaryIfaceIP)).To(Succeed())
+				Expect(inRange(ipv4TestRange, secondaryIfaceIPs[0])).To(Succeed())
+			})
+		})
+
+		Context("DualStack test (IPRanges without old range)", func() {
+			BeforeEach(func() {
+				const dualstackPodName = "whereabouts-dualstack-test"
+				var err error
+
+				netAttachDefDualStack = macvlanNetworkWithWhereaboutsIPAMNetwork(testDualStackNetworkName, testNamespace, "", testIPRangesDualStack)
+
+				By("creating DualStack NetworkAttachmentDefinition for whereabouts")
+				_, err = clientInfo.AddNetAttachDef(netAttachDefDualStack)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("creating a pod with whereabouts net-attach-def")
+				pod, err = clientInfo.ProvisionPod(
+					dualstackPodName,
+					testNamespace,
+					podTierLabel(dualstackPodName),
+					entities.PodNetworkSelectionElements(testDualStackNetworkName),
+				)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				By("deleting pod with whereabouts net-attach-def")
+				Expect(clientInfo.DeletePod(pod)).To(Succeed())
+				By("deleting DualStack NetworkAttachmentDefinition for whereabouts")
+				Expect(clientInfo.DelNetAttachDef(netAttachDefDualStack)).To(Succeed())
```

Comment:
I would remove the `By` clauses. 

They're pretty much useless here - you just want to get rid of existing entities ... What is the motivation for having these ?

Reply:
I was trying to keep the coding style consistent with other tests. For e.g.

https://github.com/k8snetworkplumbingwg/whereabouts/blob/38ded4bbc37e5205f4cefc75c6902fa414ef32d2/e2e/e2e_test.go#L109-L110